### ISSUE TITLE
Alarm repeat  problem

### ISSUE THREAD

**Issue by sizarta** [theScrabi/Terminightor#4]
Hello my friend. Tnx for your great app.
I found a problem in your app and It's very strange. when I check the repeat check box and choose some days often it didn't work or worked wrong . some times it didn't repeat , some times repeated all the weeks days and some times work fine. Can you help me?

**Comment by theScrabi** [comment]
What android version do you have? I have the suspicion that android 6 wound run this app proper. 
Even android 5 may seem to have trouble.

**Comment by sizarta** [comment]
My android version is 5 . For example today is Wednesday and I set an alarm for Friday and Sunday. the alarm don't work today and it's correct. but when I change the android date manually to Friday and Sunday it didn't work yet. but when I set an alarm repeat consist of today it works correctly. I try this for android standard alarm and it was fine.

**Comment by theScrabi** [comment]
Hm that’s strange...

**Comment by sizarta** [comment]
I hope u can solve this problem because I use it all the week. Overall it's great :+1: and tnx for creating this app :)

**Comment by theScrabi** [comment]
I understand you. I'm even reliant to it. I need this app otherwise I wount make it out of my bed early enough.

Have you tried reinstalling it?

**Comment by sizarta** [comment]
Yes . Many times. some times it works fine and some times don't work. even some times works late

**Comment by theScrabi** [comment]
Hm ok I'll se what I can do. But I can not promise that I will come up with a solution that early :/

**Comment by sizarta** [comment]
I believe you're Genius :)

**Comment by theScrabi** [comment]
I might have found the source of the error, but I'm not shure since I can't test it because everything works on all of my smartphone.

So if it still causes trouble, you might have to provide a log, or try to run Terminightor on another smartphone, to see how it behaves there.

Please try this test version. If it runs, i can release it on f-droid:
<removed>

**Comment by sizarta** [comment]
My browser did n't open this link because it has security problem.

**Comment by theScrabi** [comment]
Hm looks like this didn't solve it, I know have trouble with that new patch :/

**Comment by theScrabi** [comment]
I'm sorry I can't find the error. Debugging a timer is a mess, and everything should work according to the code. Also I can't just simply test it on different android versions, since this app doesn't run on emulators (because of the NFC). Therefore I have to stick to to my smartphone.
And yes I encountered some bugs myself, but I couldn't reproduce them.
I surrender -.-

**Comment by sizarta** [comment]
can you use the other apps time managers?
this 2 apps working correctly.
https://f-droid.org/repository/browse/?fdfilter=clock&fdid=za.co.neilson.alarm
https://f-droid.org/repository/browse/?fdfilter=clock&fdid=com.angrydoughnuts.android.alarmclock

**Comment by theScrabi** [comment]
Yes that would be an option. I'll try that, as soon as i find time for it :)

**Comment by xenithorb** [comment]
Is it as simple as disabling battery optimization?

**Comment by theScrabi** [comment]
Does it still not work?
